File: Artificial Intelligence/family tree/grandparent.py
tupleList1=[('parent', 'Hasib', 'Rakib'),('parent', 'Rakib', 'Sohel'),
             		('parent', 'Rakib', 'Rebeka'),('parent', 'Rashid', 'Hasib')]

# Procedure to find the grandparent of X

X=str(input("Grandchildren:"))
print('Grandparent:', end=' ')
i=0
while(i<=3):
    if ((tupleList1[i][0] == 'parent')&( tupleList1[i][2] == X)):
        for j in range(4):
            if ((tupleList1[j][0] == 'parent') & ( tupleList1[i][1] == tupleList1[j][2])):
                print(tupleList1[j][1], end=' ')   
    i=i+1  

File: Artificial Intelligence/family tree/grandparent.pl
parent('Hasib', 'Rakib'). parent('Rakib', 'Sohel'). parent('Rakib', 'Rebeka'). parent('Rashid', 'Hasib'). grandparent(X,Z) :- parent(X, Y), parent(Y, Z).
findGp :- write('Grandchildren : '), read(X), write('Grandparent : '),
	grandparent(Gp, X), write(Gp), tab(5), fail.
findGp.
